lib/ProductOpener/Users.pm

```diff
@@ -997,27 +1177,98 @@ sub store_user_session ($user_ref) {
 }
 
 sub store_user ($user_ref) {
-	my $userid = $user_ref->{userid};
+	my $oidc_implementation_level = get_oidc_implementation_level();
+	if ($oidc_implementation_level < 5) {
+		# We are still using legacy code for registration
+
+		if ($oidc_implementation_level > 0) {
+			# Sync the user with Keycloak
+			my $keycloak = ProductOpener::Keycloak->new();
+			$keycloak->create_or_update_user($user_ref);
```

Comment:
This could be intentional, but we should discuss it real quick.

`store_user` is called by `update_login_time`, which is called after the password was successfully validated in `init_user`. Especially for some apps and API clients, the user+password tuple is sent with every request, causing the login time to be updated on each request. `ProductOpener::Keycloak::create_or_update_user` is relatively heavy as it executes several REST requests. Of the Keycloak API endpoints, those that update user records are probably not as well optimized as something like issuing a token.

I wonder, if that might be a performance bottleneck?

Reply:
I think it should probably call store_user_session instead

Reply:
Don't get me wrong: There are advantages of calling it upon login. Namely, active users get synced to Keycloak immediately. However, I tend to agree because of the performance impact.

Reply:
I don't think we should be updating the user in Keycloak on each login, only when their details change